src/common/util/userTimeout.ts

```diff
@@ -0,0 +1,137 @@
+import { arrayPatternMatch } from './arrayPatternMatch'
+import { isStringPatternMatch } from './stringPatternMatch'
+
+/**
+ * Default Session Idle Time
+ *
+ * If the timeout configuration has not been set, or the User has no roles ( although the
+ * User shouldn't reach this point if they have no roles associated with them ) the
+ * default session idle time will be used.
+ */
+
+export const DEFAULT_SESSION_TIMEOUT = {
+    idleModalDisplayTime: 10,
+    pattern:
+        'ERROR: NO-SESSION_TIMEOUT_SET. You need to set a DEFAULT Session Timeout for this application through' +
+        'the configuration file. ie. use the pattern ".", @see unit tests. The totalIdleTime will be set to a low value.',
+    totalIdleTime: 480,
+}
+
+/**
+ * Is Role Match
+ *
+ * Checks if a User's role, matches a specified Regular Expression.
+ *
+ * We use a Regular Expression so that we can set the default session timeout via configuration ie. '.', hence we used
+ * JS .match over .includes.
+ *
+ * There will be a different default session timeout per application, and different session timeouts per user groups
+ * hence setting it via configuration.
+ *
+ * @param role - 'pui-case-manager'
+ * @param pattern - 'case-manager' / 'pui-' / '.'
+ * @returns {boolean}
+ */
+export const isRoleMatch = (role: string, pattern: string): boolean => {
+    return isStringPatternMatch(role, pattern)
+}
+
+/**
+ * Any Roles Match
+ *
+ * Checks an array of roles for pattern matches.
+ *
+ * @param roles - [
+ *  'pui-case-manager',
+ *  'pui-finance-manager',
+ * ]
+ * @param pattern - 'case-manager' / 'pui-' / '.'
+ */
+export const anyRolesMatch = (roles: string[], pattern: string): boolean => {
+    return arrayPatternMatch(roles, pattern)
+}
+
+/**
+ * Sort User Roles
+ *
+ * Should sort the User's Roles alphabetically. Why? So that a priority order can be given to the Session Timeout +
+ * configuration list.
+ *
+ * We clone the original array, so that we avoid mutation.
+ *
+ * Example: If we want a PUI Session Timeout to be given preference over another Session Timeout it would be further
+ * up the Session Timeout Configuration list.
+ *
+ * @param - [
+ * 'caseworker-divorce-financialremedy',
+ * 'pui-user-manager',
+ * 'caseworker-probate-solicitor',
+ * 'caseworker',
+ * 'caseworker-probate',
+ * 'pui-finance-manager',
+ * 'caseworker-divorce-solicitor',
+ * ]
+ * @return - [
+ * 'caseworker',
+ * 'caseworker-divorce-financialremedy',
+ * 'caseworker-divorce-solicitor',
+ * 'caseworker-probate',
+ * 'caseworker-probate-solicitor',
+ * 'pui-user-manager',
+ * 'pui-finance-manager',
+ * ]
+ */
+export const sortUserRoles = (roles: string[]) => roles.sort()
```

Comment:
this already exists as ```sortArray```

Reply:
Yep, I believe that proxying to the underlying function is better (in this case), as it allows commenting of the underlying function, to explain how it's being used in the context of this file.

Reply:
Sorry, I meant as in this function wasn't being proxied :sweat_smile:

Reply:
Also, it will fix the code smell reported by sonar

https://sonarcloud.io/project/issues?id=hmcts_rpx-xui-node-lib&open=AXMzv7WWnCarh0zG3uSz&pullRequest=93&resolved=false&types=CODE_SMELL